### ISSUE TITLE
Debugger fails to start

### ISSUE THREAD

**Issue by rossamurphy** [fatih/vim-go#3693]
### What did you do? (required: The issue will be **closed** when not provided)
I tried to start the debugger with :GoDebugStart

### What did you expect to happen?
The debugger to start

### What happened instead?
ERR: 2024-12-20T10:13:38Z debug layer=rpc API server pid = 95006
ERR: 2024-12-20T10:13:38Z info layer=debugger launching process with args: [/var/folders/cq/k7gz880x2sqd26yvf2k5vgwh0000gn/T/nvim.rossmurphy/sCdga8/1]
ERR: 2024-12-20T10:13:38Z debug layer=debugger entryPoint 0x100214000 machoOff 0x100000000
ERR: Warning: no debug info found, some functionality will be missing such as stack traces and variable evaluation.
ERR: 2024-12-20T10:13:38Z warning layer=debugger debug_frame workaround not applied: could not find a Go function
ERR: could not launch process: could not read debug info (decoding dwarf section info at offset 0x0: too short) and could not read go symbol table (could not find rodata struct member)

### Configuration (**MUST** fill this out):
Installed using Packer
	use({ 'fatih/vim-go', run = ':GoUpdateBinaries' })

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary>

```vim

```
</details>

#### Vim version (first three lines from `:version`):

NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1727870382

#### Go version (`go version`):

go version go1.22.2 darwin/arm64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=''
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/rossmurphy/Library/Caches/go-build'
GOENV='/Users/rossmurphy/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/rossmurphy/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/rossmurphy/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/Users/rossmurphy/go/go1.22.2'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/rossmurphy/go/go1.22.2/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.22.2'
GCCGO='gccgo'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='1'
GOMOD='/Users/rossmurphy/GitHub/cambot/leedor/reader/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/cq/k7gz880x2sqd26yvf2k5vgwh0000gn/T/go-build2494515860=/tmp/go-build -gno-record-gcc-switches -fno-common'

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.17.0

</pre></details>

#### vim-go configuration:
<details><summary>vim-go configuration</summary><br><pre>
g:go_loaded_gosnippets = 1
g:go_jump_to_error = 1
g:go_loaded_install = 1
</pre></details>

#### filetype detection configuration:
<details><summary>filetype detection</summary><br><pre>
filetype detection:ON  plugin:ON  indent:ON
</pre></details>

**Comment by bhcleek** [comment]
I've tried to duplicate this several times without success. I've tried both an older and a newer version of delve.

What happens when you use delve on the CLI? You can see the command and arguments that vim-go uses by first running `:let g:go_debug=['shell-commands']` and then using `:GoDebugStart`. When using delve's CLI, you'll likely want to omit at least the `--headless` and `--listen` arguments.

**Comment by rossamurphy** [comment]
thanks for coming back to me, I managed to fix this finally by updating my Go to the latest version. From what I gather, it was related to [this issue](https://github.com/go-delve/delve/issues/3797), as I had recently updated to Mac OS Sequoia. I also tried launching the debugger in IntelliJ GoLand, and in [this plugin](https://github.com/ray-x/go.nvim), so it was not your plugin after all, but, rather, a more general system-wide incompatibility between my version of Go and my OS. Thanks for taking a look and trying to help me out!
